### PR TITLE
Add TokenDownload tied to download url creation

### PR DESCRIPTION
**Pull request by davidsmejia**
## Issue Number

#963 

## Purpose/Implementation Notes

- Adds TokenDownload model
- Creates new TokenDownload record when API returns downloadable link

## Types of changes

- New feature (non-breaking change which adds functionality)

## Functional tests

- Added assertions to check that they are created and not created when expected

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Screenshots

N/A

### CODE REVIEW

**Review by nozomione — APPROVED** [review]
I've made a few small suggestions. Otherwise, this PR looks great - LGTM!

File: api/scpca_portal/models/token_download.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        db_table = "track_token_download"
+        get_latest_by = "updated_at"
+        ordering = ["updated_at"]
+
+    token = models.TextField()
+    project_id = models.TextField(null=True)
+    sample_id = models.TextField(null=True)
+    format = models.TextField(null=True)
+    modality = models.TextField(null=True)
+    includes_merged = models.BooleanField(default=False)
```
[the remark this answers is not in the document]
Comment (davidsmejia) [reply api/scpca_portal/models/token_download.py:18]: Thats a good call, no reason not to include it

File: api/scpca_portal/views/computed_file.py [hunk 2]
```diff
@@ -88,6 +89,7 @@ def get_serializer_context(self):
                     {"message": "Your token is not valid or not activated.", "token_id": token_id}
                 )
 
+            TokenDownload.track(token_id, computed_file_pk)
```
Comment (avrohomgottlieb) [inline api/scpca_portal/views/computed_file.py:92]: I would move this here for clarity, but it's not essential.

```suggestion
            computed_file_pk = self.kwargs.get("pk")
            TokenDownload.track(token_id, computed_file_pk)
```

File: api/scpca_portal/models/token_download.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    def __str__(self):
+        return f"TrackTokenDownload {self.token}"
+
+    @classmethod
+    def track(cls, token_id, computed_file_id):
+        if computed_file := ComputedFile.objects.filter(id=computed_file_id).first():
+            TokenDownload.objects.create(
+                token=token_id,
+                project_id=computed_file.project,
+                sample_id=computed_file.sample,
```
[the remark this answers is not in the document]
Comment (avrohomgottlieb) [reply api/scpca_portal/models/token_download.py:31]: If this is the case, then we should either conditionally assign these TextFields, or convert them to ForeignKeys

**Review by avrohomgottlieb — APPROVED** [review]
After correcting the id bug that @nozomione found, we're good to go!

**Pull request merged**